src/Strategist/Strategist.cpp

```diff
@@ -27,24 +27,35 @@ void Strategist::decrementSupply() {
 }
 
 void Strategist::determineMapSize() {
-    // Need to figure out how we want to determine this, or if we even want to determine it in the strategist.
-    map_size = smallest;
+    // For now, we determine map size based off # of spawn locations
+    if (BWAPI::Broodwar->getStartLocations().size() <= 4) {
+        map_size = smallest;
+    }
+    else if (BWAPI::Broodwar->getStartLocations().size() <= 6) {
+        map_size = medium;
+    }
+    else {
+        map_size = large;
+    }
 }
 
 void Strategist::swapBuildOrder() {
     // Note: swapBuildOrder() function should only be called when we first discover enemy race
 
-    int dronesMorphed = 0;
     std::queue<std::pair<BWAPI::UnitType, int>> newBuildQueue;
-
-    // UnitType key, int value
-    // Value tracks the count of those units we currently possess
-    std::map<BWAPI::UnitType, int> unitsByCount = player.getUnitCount();
+    auto unitsByCount = player.getUnitCount();
 
     // Adjust for starting units
     unitsByCount[BWAPI::UnitTypes::Zerg_Drone] -= 4;
     unitsByCount[BWAPI::UnitTypes::Zerg_Hatchery] -= 1;
 
+    // Adjust for drones morphed into buildings
+    for (auto& [key, value] : unitsByCount) {
+        if (key.whatBuilds().first == BWAPI::UnitTypes::Zerg_Drone) {
```

Comment:
All buildings are initially made from `BWAPI::UNitTypes::Zerg_Drone`, but some buildings are made from other buildings. We can instead check if `key.isBuilding()` is true instead, since this will tell us if a Drone was morph regardless of what building it is.